src/classes/FeatureManagementService.cls

```diff
@@ -0,0 +1,29 @@
+public with sharing class FeatureManagementService {
+    List<FeatureParam> recordCounts = new List<FeatureParam>{
+        new FeatureParam('JobsAllTime').withRecordCount(Volunteer_Job__c.SObjectType, ''),
+        new FeatureParam('JobsLastYear').withRecordCount(Volunteer_Job__c.SObjectType, 'CreatedDate = LAST_YEAR'),
+        new FeatureParam('JobsThisYear').withRecordCount(Volunteer_Job__c.SObjectType, 'CreatedDate < THIS_YEAR')
+    };
+
+    List<FeatureParam> fieldCounts = new List<FeatureParam>{
+        new FeatureParam('JobsCustomFields').withCustomFieldCount(Volunteer_Job__c.SObjectType),
+        new FeatureParam('ShiftsCustomFields').withCustomFieldCount(Volunteer_Shift__c.SObjectType),
+        new FeatureParam('HoursCustomFields').withCustomFieldCount(Volunteer_Hours__c.SObjectType)
+    };
+
+    public FeatureManagementService() {}
+
+    public void updateRecordCounts() {
+        for (FeatureParam recordCount : recordCounts) {
+            recordCount.send();
```

Comment:
I love the overall structure of this, but where you might run into issues is trying to do all the count queries in a single execution context. I addressed this in NPSP by breaking up the feature parameter data gathering into a batch job that iterates over a List<String>, allowing the count queries to be executed in independent batch iterations.

Reply:
Thanks for the feedback, definitely something we'll need to keep in mind in the design as well.